tests/webservice_test.py

```diff
@@ -216,96 +211,86 @@ def test_300_error(self):
         self._test_error(400, "IP_ADDRESS_RESERVED", AddressNotFoundError)
 
-    @httprettified
     def test_permission_required(self):
         self._test_error(403, "PERMISSION_REQUIRED", PermissionRequiredError)
 
-    @httprettified
     def test_auth_invalid(self):
         self._test_error(400, "AUTHORIZATION_INVALID", AuthenticationError)
 
-    @httprettified
     def test_license_key_required(self):
         self._test_error(401, "LICENSE_KEY_REQUIRED", AuthenticationError)
 
-    @httprettified
     def test_account_id_required(self):
         self._test_error(401, "ACCOUNT_ID_REQUIRED", AuthenticationError)
 
-    @httprettified
     def test_user_id_required(self):
         self._test_error(401, "USER_ID_REQUIRED", AuthenticationError)
 
-    @httprettified
     def test_account_id_unkown(self):
         self._test_error(401, "ACCOUNT_ID_UNKNOWN", AuthenticationError)
 
-    @httprettified
     def test_user_id_unkown(self):
         self._test_error(401, "USER_ID_UNKNOWN", AuthenticationError)
 
-    @httprettified
     def test_out_of_queries_error(self):
         self._test_error(402, "OUT_OF_QUERIES", OutOfQueriesError)
 
     def _test_error(self, status, error_code, error_class):
         msg = "Some error message"
         body = {"error": msg, "code": error_code}
-        httpretty.register_uri(
-            httpretty.GET,
-            self.base_uri + "country/1.2.3.18",
-            body=json.dumps(body),
+        self.httpserver.expect_request(
+            "/geoip/v2.1/country/" + "1.2.3.18"
+        ).respond_with_json(
+            body,
             status=status,
             content_type=self._content_type("country"),
         )
         with self.assertRaisesRegex(error_class, msg):
             self.run_client(self.client.country("1.2.3.18"))
 
-    @httprettified
     def test_unknown_error(self):
         msg = "Unknown error type"
         ip = "1.2.3.19"
         body = {"error": msg, "code": "UNKNOWN_TYPE"}
-        httpretty.register_uri(
-            httpretty.GET,
-            self.base_uri + "country/" + ip,
-            body=json.dumps(body),
+        self.httpserver.expect_request("/geoip/v2.1/country/" + ip).respond_with_json(
+            body,
             status=400,
             content_type=self._content_type("country"),
         )
         with self.assertRaisesRegex(InvalidRequestError, msg):
             self.run_client(self.client.country(ip))
 
-    @httprettified
     def test_request(self):
-        httpretty.register_uri(
-            httpretty.GET,
-            self.base_uri + "country/" + "1.2.3.4",
-            body=json.dumps(self.country),
-            status=200,
-            content_type=self._content_type("country"),
-        )
-        self.run_client(self.client.country("1.2.3.4"))
-        request = httpretty.last_request
+
+        request = None
+
+        def custom_handler(r):
+            nonlocal request
+            request = r
+            return ""
```

Comment:
The custom handler is a bit confusing. I think we could eliminate it by using `headers` or `header_value_matcher` to match the expected headers on `expect_request`. I don't think we need to test the path again.